disco_aws_automation/asiaq_cli.py

```diff
@@ -103,7 +104,125 @@ def _update_s3_configs(self, sandbox_name):
             s3_bucket.upload_file(local_config_path, remote_config_path)
 
 
+class DataPipelineCommand(CliCommand):
+    """
+    CliCommand implementation for managing data pipelines (excessively entangled with dyanmodb)
+    """
+    DESCRIPTION = "Manage data pipelines for dynamodb backup and/or other purposes."
+
+    BACKUP_TEMPLATE = 'dynamodb_backup'
+    RESTORE_TEMPLATE = 'dynamodb_restore'
+
+    @classmethod
+    def init_args(cls, parser):
+        subsub = parser.add_subparsers(title="data pipeline commands", dest="dp_command")
+        backup_parser = subsub.add_parser("configure_dynamodb_backup",
+                                          help="Configure backup for a dynamodb table")
+        backup_parser.add_argument("table_name")
+        restore_parser = subsub.add_parser("restore_dynamodb", help="Restore a dynamodb table from backup")
+        restore_parser.add_argument("table_name")
+        restore_parser.add_argument("--from", dest="backup_dir",
+                                    help="Previous backup to restore from (default: latest)")
+        list_parser = subsub.add_parser("list", help="List available pipelines")
+        list_parser.add_argument("--pipeline-name", dest="search_name", help="Find pipelines with this name.")
+        list_parser.add_argument("--all-envs", dest="ignore_env", action='store_const', const=True)
+
+    def run(self):
+        dispatch = {
+            'list': self._search,
+            'configure_dynamodb_backup': self._create_backup,
+            'restore_dynamodb': self._restore_backup,
```

Comment:
Actually I see support for delete in the code below so presumably just not added at this level yet.  Cool.